presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,518 @@
+    {
+        return deserialize(input).dimension();
+    }
+
+    @SqlNullable
+    @Description("Returns true if and only if the line is closed")
+    @ScalarFunction("st_is_closed")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stIsClosed(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "st_is_closed", LINE_STRING, MULTI_LINE_STRING);
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        int pathCount = lines.getPathCount();
+        boolean result = true;
+        for (int i = 0; result && i < pathCount; i++) {
+            Point start = lines.getPoint(lines.getPathStart(i));
+            Point end = lines.getPoint(lines.getPathEnd(i) - 1);
+            result = result && end.equals(start);
+        }
+        return result;
+    }
+
+    @SqlNullable
+    @Description("Returns true if and only if the geometry is empty")
+    @ScalarFunction
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stIsEmpty(@SqlType(NAME) Slice input)
+    {
+        return deserialize(input).isEmpty();
+    }
+
+    @Description("Returns length of the geometry if it is a linestring or multilinestring")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stLength(@SqlType(NAME) Slice input)
+    {
+        return deserialize(input).getEsriGeometry().calculateLength2D();
+    }
+
+    @Description("Returns the maximum X coordinate of geometry")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stMaxX(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getXMax();
+    }
+
+    @Description("Returns the maximum Y coordinate of geometry")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stMaxY(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getYMax();
+    }
+
+    @Description("Returns the minimum X coordinate of geometry")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stMinX(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getXMin();
+    }
+
+    @Description("Returns the minimum Y coordinate of geometry")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stMinY(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getYMin();
+    }
+
+    @Description("Returns the number of interior rings in the polygon")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long stInteriorRingNumber(@SqlType(NAME) Slice input)
```

Comment:
Should this also be *count* instead of *number*?

Reply:
FWIW, PostGIS version is named ST_NumInteriorRings: http://postgis.org/docs/ST_NumInteriorRings.html